docs/decisions/20230928-revocation-status-list-expansion-strategy.md

```diff
@@ -0,0 +1,74 @@
+
+## Context and Problem Statement
+
+In the process of devising a mechanism for revoking JWT credentials, we've opted to implement the [statusList2021](https://www.w3.org/TR/vc-status-list/) method. 
+This approach entails the creation of a "status list credential" that incorporates a gZip compressed status list in a form a bitString.
+The specification recommends a minimum size of 16 KB for the status list included in a credential.
+However, it does not delineate a maximum size, nor does it provide guidance on how to proceed if the selected status list surpasses its capacity to store information about revoked credentials.
+Put differently, if more credentials are issued than can be accommodated by a 16 KB status list, no specific instructions are provided.
+
+
+## Decision Drivers
+
+We must determine a strategy for expanding the revocation status list to accommodate the increasing number of revoked credentials in the future.
+
+It's crucial to keep in mind that this status list will be part of a "status list credential."
+This credential will be frequently requested through the REST API during verification by verifiers and will be downloaded over the network.
+Therefore, we need to ensure that the status list remains reasonably small in size to prevent any slowdowns in the verification process.
+
+In the future, there might be a need to reorganize the state and possibly merge various status lists into a single comprehensive registry for verifiers to depend on. This is not the current scenario, as each Prism agent currently maintains status lists specific to their respective tenants.
+
+Absolutely, it's crucial to avoid overengineering the solution. This ensures that the code remains manageable and easy to maintain in the long run.
+
+
+## Considered Options
+
+
+Option 1: Increment status list size as we approach its limit:
+
+We'll enhance the status list by simply doubling its size.
+If we opt for the minimum size of 16 KB for the revocation status list, it can manage up to 131,072 revocable credentials before requiring expansion.
+It's important to note that this number represents the potential capacity for revocable credentials to be issued, not necessarily the actual number of credentials that have been revoked.
+Even unrevoked credentials still occupy space.
+
+Option 2: Generate multiple revocation status lists as the previous one reaches its limit
+
+With this approach, we'll generate and store multiple status list credentials.
+It will be crucial to ensure that each credential is linked to a specific status list, allowing us to track where the revocation information is stored.
+If we stick with the smallest recommended status list size, one revocation status list can hold information about 131,072 revocable credentials.
+
+
+## Decision Outcome
+
+
+### Positive Consequences
+
+
+### Negative Consequences
+
+
+## Pros and Cons of the Options
+
+Option 1 offers the primary advantage of being straightforward to implement.
```

Comment:
beyond simplicity there is a privacy aspect
It is true that this revocation mechanism is not fully private, but there is one privacy property it tries to enforce. Namely that the issuer should not infer which VC is being verified by a verifier.
In that aspect, option 2 does diminishes privacy compared to option 1 in some scenarios. For instance, imagine a use case where VCs are created with low frequency, but the number of VCs always grows. At the very beginning, there is a small number of VCs, which makes the anonymity set of the status list small (even though it can have a size of 130K VCs, the issuer may have issued just a few). This issue is the same for option 1 and 2.
Now, at some point, the number of VCs will be big enough to make a reasonable anonymity set size. However, in option 2, when the issuer reaches the limit of 16Kb and creates a new list, there will be a period of time where there will be few VCs in the new list. This would make the anonymity set small for the VCs in the second list. This is different from option 1, where the anonymity set keeps growing.
This is not the only criteria to consider, but would be good to list in the document.
I see the two scenarios as alternatives for different situations. If the expected number of VCs is not going to reach a point that makes the status list "too big" (a metric to determine on performance tests), then option 1 is good. But, if we are talking of a number of VCs so big that performance is an issue, then option 2 could suit better. (or a mix of the two, where you double sizes up to certain limit, and then just create a new list)